subprojects/docs/src/samples/userguide/dependencyManagement/workingWithDependencies/iterateDependencies/build.gradle

```diff
@@ -12,12 +12,14 @@ dependencies {
 }
 
 //START SNIPPET iteration-task
-task iterateScmDependencies {
+task iterateDeclaredDependencies {
     dependsOn configurations.scm
```

Comment:
❌ This dependsOn is counterproductive. Since the task only iterates over the declared dependencies, it doesn't need the configuration to be resolved before. This will just make this task run slower than it needs to.

Reply:
I made this change based on a comment from a previous review: https://github.com/gradle/gradle/pull/4124#discussion_r163356699. It's somewhat not relevant to the example. However, I know that a lot of people just copy/paste whatever snippet they find on the web and then complain that it doesn't work properly in other contexts.

Reply:
That was for iterating the artifacts. But herr you just want declared dependencies. Depending in the configuration would download the artifacts even though your task doesn't need them at all. Maybe replace this with an example that isn't a task to make clear that this is safe to call at any time?

Reply:
Removed. I added a sentence which clearly states that it is safe to call this during configuration time.